pkg/controller/storagecluster/reconcile.go

```diff
@@ -1152,3 +1156,129 @@ func getMonCount(nodeCount int) int {
 
 	return count
 }
+
+//ensureJobTemplates returns spec of a job template
+func (r *ReconcileStorageCluster) ensureJobTemplates(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	osdCleanUpTemplate := &openshiftv1.Template{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sc.Name + "-osd-removal",
+			Namespace: sc.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, osdCleanUpTemplate, func() error {
```

Comment:
Can we use `r.client.Create(context.TODO(), types.NamespacedName{...}, osdCleanUpTemplate)` for simplicity?

Reply:
We should do proper reconciliation and make sure we either Create or Update as appropriate. This seems like an appropriate use of this helper function, and we should probably consider using it elsewhere... unless there's a reason not to? The alternative is certainly more complicated: https://github.com/openshift/ocs-operator/blob/master/pkg/controller/storagecluster/reconcile.go#L727-L763

Reply:
Umm I would still prefer this createOrUpdate method as this has been a recommended way of creating/updating the resources. https://godoc.org/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#CreateOrUpdate